### PR TITLE
Add support for returning location of schema elements from the PDL schema parser.

### PR DESCRIPTION
**Pull request by karthikbalasub**
This feature would be useful for tools that provide additional functionality like linting, code search etc on the PDL schemas.

### CODE REVIEW

File: data/src/main/java/com/linkedin/data/schema/grammar/PdlSchemaParser.java [hunk 1]
```diff
@@ -489,9 +583,24 @@ else if (namedType.enumDeclaration() != null)
...
 
+  private void recordLocation(Object schemaNode, ParserRuleContext context)
+  {
+    if (_isLocationNeeded)
+    {
+      // getCharPosition returns beginning of the last token. Add the token's length to get actual end position.
+      int endPosition = context.getStop().getCharPositionInLine() +
+          (context.getStop().getStopIndex() - context.getStop().getStartIndex());
+      // Parser columns are indexed at 0, so add 1 to get it 1 indexed.
+      _parseLocations.put(schemaNode,
```
Comment (BrianPin) [inline data/src/main/java/com/linkedin/data/schema/grammar/PdlSchemaParser.java:598]: So what is a schemaNode? 
Is it the whole PDL file record or the smallest unit on the schema (like the field)
If it is the whole file then there seems to be only one location allowed to be recorded?

Comment (karthikbalasub) [reply data/src/main/java/com/linkedin/data/schema/grammar/PdlSchemaParser.java:598]: schema node (I'll rename it as element) is any "referencable" element in the schema. It includes fields, union members, inline schemas, enum symbols etc. For the full list, look at the java doc of getParseLocations method.

**Review by BrianPin — APPROVED** [review]
LGTM

**Review by nickibi — APPROVED** [review]
LGTM

**Pull request merged**